Source/items.cpp

```diff
@@ -4153,12 +4155,12 @@ int RndVendorItem(int minlvl, int maxlvl)
 			break;
 	}
 
-	return ril[GenerateRnd(ri)] + 1;
+	return static_cast<_item_indexes>(ril[GenerateRnd(ri)] + 1);
 }
 
-int RndSmithItem(int lvl)
+_item_indexes RndSmithItem(int lvl)
 {
-	return RndVendorItem<SmithItemOk, true>(0, lvl);
+	return static_cast<_item_indexes>(RndVendorItem<SmithItemOk, true>(0, lvl) - 1);

```

Comment:
Is it not possible to have this return the enum? How about a struct that can also represent the book that would normally be represented via negative numbers

Reply:
Not really, because of the weird subtraction. Even if `RndVendorItem` returns the enum (which it does) the subtraction forces we to convert again.